### PR TITLE
Skip failing tests in test_nn if compiled without LAPACK

### LINKED ISSUE #70912: Tests in test_nn fail when compiled without LAPACK support
### 🐛 Describe the bug

Running python `test/run_test.py` fails in `test_nn` module when PyTorch is compiled w/o LAPACK support.

The expected behavior is to skip tests for LAPACK-relative functionality in such case.

The full error log is [here](https://gist.github.com/lithuak/ec5f36059015f7c9b33cd4c50efad06e):

My proposition is to fix this by adding the `@skipIfNoLapack` decorator to failing test as itis already done with some other test in the same module. 

I'll add the PR with proposed fix.

### Versions

Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.22.0

Python version: 3.8 (64-bit runtime)
Is CUDA available: N/A
CUDA runtime version: 10.2.89
GPU models and configuration: GPU 0: GeForce RTX 2080 Ti
Nvidia driver version: 460.91.03
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.21.4
[pip3] pytorch-sphinx-theme==0.0.24
[pip3] torch==1.11.0a0+gitd1dbcb1
[pip3] torchvision==0.11.1
[conda] Could not collect

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345

### PR DESCRIPTION
**Pull request by lithuak**
Fixes #70912

### CODE REVIEW

**Comment by albanD** [comment]
cc @Lezcano these tests are important. We should make sure we can run them without Lapack. Should we pick a different set of functions to ensure it is true?

**Comment by lithuak** [comment]
@albanD I'm new to the codebase, so my proposed fix was just an improvised attempts to make it work.

One small thing that comes to mind - use `@skipCPUIfNoLapack` instead of `@skipIfNoLapack`. That would make it a bit more narrow. Not sure it solves the problem though.

**Comment by albanD** [comment]
No problem!
We don't have a CI build without LAPACK unfortunately as this is something that is available in the vast majority of cases. So maybe this fix is good since it doesn't change anything on the actual CI.
I just want the opinion from @Lezcano on that. You can leave the PR as-is for now.

**Comment by Lezcano** [comment]
I agree that we should use a simpler function to test this functionality that does not depend on LAPACK. 

I think that this skip is fine for now given that we always use LAPACK in the CI, but it'd be good to at least leave a `# FIXME`  comment over these tests explaining what's going on here.

**Comment by albanD** [comment]
Sounds good!
@lithuak could you add the #FIXME comment next the skips please and I'll land that!

**Comment by lithuak** [comment]
@yep. I think I can also add a github issue listing those tests to be fixed WDYT

**Comment by albanD** [comment]
Yes, an issue is even better!

File: test/test_nn.py [hunk 1]
```diff
@@ -4807,6 +4825,9 @@ def assert_weight_allclose_Q(weight, W):
                     opt.step()
                     assert_is_orthogonal(m.weight)
 
+    # FIXME: Rewrite this test using functions not depending on LAPACK
+    #        and remove the `@skipIfNoLapack` (see #70995)
```
Comment (Lezcano) [inline test/test_nn.py:4829]: This one (and the other orthogonal test) are expected to depend on LAPACK, so the FIXME doesn't apply to them.

**Review by lezcano — APPROVED** [review]
LGTM modulo a very small nit!

**Review by lithuak — COMMENTED** [review]
started a review by mistake, pls ignore it

**Review by albanD — APPROVED** [review]
Thanks for the update!

**Pull request closed without merging**